### PR TITLE
Implement font-scale, line-height and vertical-rythm scales

### PR DESCRIPTION
**Pull request by helenemccarron**
This is the MVP for font scales etc. See example of use in the index files. This should allow everyone to start using the scales while I work on replacing all the not-implemented occurrences.

### CODE REVIEW

File: styles/designs/common/pdf/typography.scss [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  $baseLineHeight: $baseFontSize * $lineHeightModifier;
+  $cellHeight: $baseLineHeight * $granularity;
+  $numCells: round($baseLineHeight * pow($scaleMultiplier, $scaleN) / $cellHeight);
+  $actualLineHeight: $numCells * $cellHeight;
+  $actualFontSize: $actualLineHeight / $lineHeightModifier;
+
+  @return $actualLineHeight + "rem";
+}
+
+@function vertical-rythm($scaleN) {
```
Comment (tomjw64) [inline styles/designs/common/pdf/typography.scss]:
```suggestion
@function vertical-rhythm($scaleN) {
```

File: styles/designs/common/pdf/typography.scss [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  $baseFontSize: 1; //rem
+  $lineHeightModifier: 1.5;
+  $granularity: 0.1;
+  $scaleMultiplier: 1.2;
+  $baseLineHeight: $baseFontSize * $lineHeightModifier;
+  $cellHeight: $baseLineHeight * $granularity;
+  $numCells: round($baseLineHeight * pow($scaleMultiplier, $scaleN) / $cellHeight);
+  $actualLineHeight: $numCells * $cellHeight;
+  $actualFontSize: $actualLineHeight / $lineHeightModifier;
```
Comment (tomjw64) [inline styles/designs/common/pdf/typography.scss]: Lines 73-75 are unused.

File: styles/designs/common/pdf/typography.scss [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  @return $actualLineHeight + "rem";
+}
+
+@function vertical-rythm($scaleN) {
+  $baseFontSize: 1; //rem
+  $lineHeightModifier: 1.5;
+  $granularity: 0.1;
+  $scaleMultiplier: 1.2;
```
Comment (tomjw64) [inline styles/designs/common/pdf/typography.scss]: Unused

File: styles/designs/common/pdf/typography.scss [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+@function line-height($scaleN) {
+  $baseFontSize: 1; //rem
+  $lineHeightModifier: 1.5;
+  $granularity: 0.1;
+  $scaleMultiplier: 1.2;
+  $baseLineHeight: $baseFontSize * $lineHeightModifier;
+  $cellHeight: $baseLineHeight * $granularity;
+  $numCells: round($baseLineHeight * pow($scaleMultiplier, $scaleN) / $cellHeight);
+  $actualLineHeight: $numCells * $cellHeight;
+  $actualFontSize: $actualLineHeight / $lineHeightModifier;
```
Comment (tomjw64) [inline styles/designs/common/pdf/typography.scss]: Unused

File: styles/templates/template1/parts/_index-settings.scss [hunk 5]
```diff
@@ -7,39 +7,43 @@
...
             column-count: 3,
             column-gap: not-implemented(5rem),
-            line-height: not-implemented(2rem),
         ),
         TitleContainer: (
             border-bottom-color: (_ref: 'colorMap:::fontBodyColor'),
             border-bottom-width: not-implemented(0.1rem),
             border-bottom-style: solid,
-            padding-bottom: not-implemented(1rem),
+            padding-bottom: vertical-rythm(2),
```
Comment (tomjw64) [inline styles/templates/template1/parts/_index-settings.scss]:
```suggestion
            padding-bottom: vertical-rhythm(2),
```

File: styles/templates/template1/parts/_index-settings.scss [hunk 6]
```diff
@@ -7,39 +7,43 @@
...
+            line-height: line-height(5),
         ),
         Group: (
             color: (_ref: 'colorMap:::fontBodyColor'),
             font-family: (_ref: 'typography:::baseFont'),
-            font-size: not-implemented(1.44rem),
-            padding-bottom: not-implemented(3rem),
+            font-size: font-scale(3),
+            line-height: line-height(3),
+            padding-bottom: vertical-rythm(3),
```
Comment (tomjw64) [inline styles/templates/template1/parts/_index-settings.scss]:
```suggestion
            padding-bottom: vertical-rhythm(3),
```

File: styles/designs/common/pdf/typography.scss [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  $scaleMultiplier: 1.2;
+  $baseLineHeight: $baseFontSize * $lineHeightModifier;
+  $cellHeight: $baseLineHeight * $granularity;
+  $numCells: round($baseLineHeight * pow($scaleMultiplier, $scaleN) / $cellHeight);
+  $actualLineHeight: $numCells * $cellHeight;
+
+  @return $actualLineHeight + "rem";
+}
+
+@function vertical-rhythm($scaleN) {
```
Comment (brittany-johnson) [inline styles/designs/common/pdf/typography.scss]: Just a question: 
Vertical rhythm will replace our values for padding and margin?

File: styles/designs/common/pdf/typography.scss [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  $baseFontSize: 1; //rem
+  $lineHeightModifier: 1.5;
+  $granularity: 0.1;
```
Comment (brittany-johnson) [inline styles/designs/common/pdf/typography.scss]: What does this mean?

File: styles/designs/common/pdf/typography.scss [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  $baseLineHeight: $baseFontSize * $lineHeightModifier;
+  $cellHeight: $baseLineHeight * $granularity;
```
Comment (brittany-johnson) [inline styles/designs/common/pdf/typography.scss]: Is cell height referring to the height of the element?

File: styles/designs/common/pdf/typography.scss (hunk 7 above)
Comment (helenemccarron) [reply styles/designs/common/pdf/typography.scss]: yes, padding-bottom and margin-bottom
padding-left, padding-right, margin-left and margin-right will use horizontal-rythm (still have to finish that one up, will do next)

File: styles/designs/common/pdf/typography.scss (hunk 8 above)
Comment (helenemccarron) [reply styles/designs/common/pdf/typography.scss]: That is the size of each unit within each cell. In essence, a cell is created based on the font size then divided into equal parts. Granularity defines how small that unit is.

Comment (tomjw64) [reply styles/designs/common/pdf/typography.scss]: The math here is to split up the entire page into rows. Granularity is a percentage of the line height which represents the height of these rows.

File: styles/designs/common/pdf/typography.scss (hunk 9 above)
Comment (tomjw64) [reply styles/designs/common/pdf/typography.scss]: Cell height == row height. See above

Comment (helenemccarron) [reply styles/designs/common/pdf/typography.scss]: Yes, each cell is a certain height. This applies to fonts, line-height and vertical spacing.

File: styles/designs/common/pdf/typography.scss (hunk 7 above)
Comment (tomjw64) [reply styles/designs/common/pdf/typography.scss]: Hopefully? ... ? most of the time? :crossed_fingers:

**Pull request merged**